File: Hangman.java
import java.util.*;

public class Hangman {
    public static void main(String[] args) {
        System.out.println("H A N G M A N");
        System.out.println();
        String[] words = {"python", "java", "kotlin", "javascript", "html", "css", "ruby",
                "sql", "coffeescript", "crystal", "elixir", "go", "groovy", "php",
                "racket", "rust", "shell", "swift", "typescript", "pascal"};
        int wins = 0;
        int loses = 0;
        Scanner scanner = new Scanner(System.in);

        label:
        while (true) {
            System.out.print("Type \"play\", \"score\", or \"exit\": ");
            String menu = scanner.nextLine();
            switch (menu) {
                case "play":
                    String currentAction = words[new Random().nextInt(words.length)];
                    char[] tip = new char[currentAction.length()];
                    Arrays.fill(tip, '-');
                    int lives = 8;
                    List<Character> usedLetters = new ArrayList<>();

                    while (true) {
                        System.out.println(String.valueOf(tip));
                        System.out.print("Input a letter: ");
                        String action = scanner.nextLine();

                        if (action.length() != 1) {
                            System.out.println("You should input a single letter");
                            System.out.println();
                            continue;
                        }
                        char letter = action.charAt(0);
                        if (!Character.isLowerCase(letter)) {
                            System.out.println("It is not an ASCII lowercase letter");
                            System.out.println();
                            continue;
                        }
                        if (usedLetters.contains(letter)) {
                            System.out.println("You already typed this letter");
                            System.out.println();
                            continue;
                        }
                        boolean letterFound = false;
                        for (int i = 0; i < currentAction.length(); i++) {
                            if (currentAction.charAt(i) == letter) {
                                tip[i] = letter;
                                letterFound = true;
                            }
                        }
                        if (!letterFound) {
                            System.out.println("No such letter in the word");
                            lives--;
                            if (lives == 0) {
                                break;
                            }
                            System.out.println("Lives left: " + lives);
                        }
                        usedLetters.add(letter);
                        if (String.valueOf(tip).equals(currentAction)) {
                            break;
                        }
                        System.out.println();
                    }
                    if (lives == 0) {
                        System.out.println("You are hanged!");
                        loses++;
                    } else {
                        System.out.println("You guessed the word " + currentAction + "!");
                        System.out.println("You survived!");
                        wins++;
                    }
                    break;
                case "exit":
                    break label;
                case "score":
                    if (wins == 1 && loses == 1) {
                        System.out.println("You won 1 time. You lost 1 time.");
                    } else if (wins == 1) {
                        System.out.println("You won 1 time. You lost " + loses + " times.");
                    } else if (loses == 1) {
                        System.out.println("You won " + wins + " times. You lost 1 time.");
                    } else {
                        System.out.println("You won " + wins + " times. You lost " + loses + " times.");
                    }
                    break;
                default:
                    System.out.println("Wrong input. Please, try again");
                    break;
            }
        }
    }
}

File: hangman.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>
#include <time.h>

#define WORDS_COUNT 20
#define MAX_WORD_LENGTH 13
#define ASCII_LOWERCASE_LENGTH 26

char words[WORDS_COUNT][MAX_WORD_LENGTH] = {
    "python", "java", "kotlin", "javascript", "html", "css", "ruby",
    "sql", "coffeescript", "crystal", "elixir", "go", "groovy", "php",
    "racket", "rust", "shell", "swift", "typescript", "pascal"
};

int wins = 0;
int loses = 0;

int is_ascii_lowercase(char c) {
    return (c >= 'a' && c <= 'z');
}

void print_score() {
    if (wins == 1 && loses == 1) {
        printf("You won 1 time. You lost 1 time.\n");
    } else if (wins == 1) {
        printf("You won 1 time. You lost %d times.\n", loses);
    } else if (loses == 1) {
        printf("You won %d times. You lost 1 time.\n", wins);
    } else {
        printf("You won %d times. You lost %d times.\n", wins, loses);
    }
}

int main() {
    srand(time(NULL));

    printf("H A N G M A N\n\n");

    while (1) {
        printf("Type \"play\", \"score\", or \"exit\": ");
        char menu[10];
        scanf("%s", menu);

        if (strcmp(menu, "play") == 0) {
            char currentAction[MAX_WORD_LENGTH];
            strcpy(currentAction, words[rand() % WORDS_COUNT]);
            char tip[MAX_WORD_LENGTH];
            int i;
            for (i = 0; i < strlen(currentAction); i++) {
                tip[i] = '-';
            }
            tip[i] = '\0';

            int lives = 8;
            char usedLetters[MAX_WORD_LENGTH] = {0};

            while (1) {
                printf("%s\n", tip);
                printf("Input a letter: ");
                char action[2];
                scanf("%s", action);

                if (strlen(action) != 1) {
                    printf("You should input a single letter\n\n");
                    continue;
                }
                char letter = action[0];
                if (!is_ascii_lowercase(letter)) {
                    printf("It is not an ASCII lowercase letter\n\n");
                    continue;
                }
                if (strchr(usedLetters, letter) != NULL) {
                    printf("You already typed this letter\n\n");
                    continue;
                }

                int letterFound = 0;
                for (i = 0; i < strlen(currentAction); i++) {
                    if (currentAction[i] == letter) {
                        tip[i] = letter;
                        letterFound = 1;
                    }
                }

                if (!letterFound) {
                    printf("No such letter in the word\n");
                    lives--;
                    if (lives == 0)
                        break;
                    printf("Lives left: %d\n", lives);
                }

                usedLetters[strlen(usedLetters)] = letter;

                if (strcmp(tip, currentAction) == 0)
                    break;

                printf("\n");
            }

            if (lives == 0) {
                printf("You are hanged!\n");
                loses++;
            } else {
                printf("You guessed the word %s!\n", currentAction);
                printf("You survived!\n");
                wins++;
            }

        } else if (strcmp(menu, "exit") == 0) {
            break;
        } else if (strcmp(menu, "score") == 0) {
            print_score();
        } else {
            printf("Wrong input. Please, try again\n");
        }
    }

    return 0;
}
